### PR TITLE
add ace medical painkillers

### PR DESCRIPTION
**Pull request by Drofseh**
- add painkillers to ace medical
- reduce pain, but only 1/8th as much as morphine
- safer to use, doesn't reduce HR, but can still OD if too many are taken too quickly (25 doses within 20 minutes)
- can only be used when conscious

### CODE REVIEW

File: addons/ace_tweaks/CfgWeapons.hpp [hunk 1]
```diff
@@ -12,4 +12,18 @@ class CfgWeapons {
 			mass = 8;
 		};
 	};
+    class CBA_MiscItem_ItemInfo;
+    class ACE_ItemCore;
+    class ACE_painkillers : ACE_ItemCore {
+        scope = 2;
+        author = "Drofseh";
+        displayName = Painkillers;
+        picture = "\x\arc_misc\addons\ace_tweaks\ui\items\painkillers_ca.paa";
```
Comment (Freddo3000) [inline addons/ace_tweaks/CfgWeapons.hpp:21]: QPATHTOF()

File: addons/ace_tweaks/CfgWeapons.hpp [hunk 2]
```diff
@@ -12,4 +12,18 @@ class CfgWeapons {
 			mass = 8;
 		};
 	};
+    class CBA_MiscItem_ItemInfo;
+    class ACE_ItemCore;
+    class ACE_painkillers : ACE_ItemCore {
+        scope = 2;
+        author = "Drofseh";
+        displayName = Painkillers;
```
Comment (Freddo3000) [inline addons/ace_tweaks/CfgWeapons.hpp:20]: No stringtable?

File: addons/ace_tweaks/CfgWeapons.hpp [hunk 3]
```diff
@@ -12,4 +12,18 @@ class CfgWeapons {
 			mass = 8;
 		};
 	};
+    class CBA_MiscItem_ItemInfo;
+    class ACE_ItemCore;
+    class ACE_painkillers : ACE_ItemCore {
```
Comment (Freddo3000) [inline addons/ace_tweaks/CfgWeapons.hpp:17]: Use GVAR here, don't mix up the ACE prefix.

File: addons/ace_tweaks/stringtable.xml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project name="ARC">
+    <Package name="Medical">
+        <Key ID="STR_ARC_Medical_Give_Painkilllers">
```
Comment (Freddo3000) [inline addons/ace_tweaks/stringtable.xml:4]: Wrong prefix

File: addons/ace_tweaks/CfgWeapons.hpp [hunk 5]
```diff
@@ -12,4 +12,18 @@ class CfgWeapons {
...
+        scope = 2;
+        author = "Drofseh";
+        displayName = Painkillers;
+        picture = "\x\arc_misc\addons\ace_tweaks\ui\items\painkillers_ca.paa";
+        model = "\A3\Structures_F_EPA\Items\Medical\PainKillers_F.p3d";
+        descriptionShort = "$STR_ARC_Medical_Treatment_Painkillers_Desc_Short";
+        descriptionUse = "$STR_ARC_Medical_Treatment_Painkillers_Desc_Use";
```
Comment (Freddo3000) [inline addons/ace_tweaks/CfgWeapons.hpp:24]: CSTRING()

File: addons/ace_tweaks/config.cpp [hunk 6]
```diff
@@ -29,6 +29,7 @@ class ACE_Medical_StateMachine {
 class ace_medical_replacementItems {
 	ItemType_401[] = {
 		{"ACE_fieldDressing", 5},
+        {"ACE_painkillers", 1},
```
Comment (Freddo3000) [inline addons/ace_tweaks/config.cpp:32]: Must bump FAK item size

File: addons/ace_tweaks/CfgVehicles.hpp [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        model = "\A3\Structures_F_EPA\Items\Medical\PainKillers_F.p3d";
+    };
+
+    // Treatment items
+    class Item_Base_F;
+    class ACE_painkillersItem: Item_Base_F {
+        scope = 2;
+        scopeCurator = 2;
+        displayName = Painkillers;
+        author = "Wilhelm Haas";
```
Comment (Freddo3000) [inline addons/ace_tweaks/CfgVehicles.hpp:19]: Who is this?

File: addons/ace_tweaks/CfgVehicles.hpp [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    class ACE_MedicalLitter_painkillers: ACE_MedicalLitterBase {
+        model = "\A3\Structures_F_EPA\Items\Medical\PainKillers_F.p3d";
+    };
+
+    // Treatment items
+    class Item_Base_F;
+    class ACE_painkillersItem: Item_Base_F {
+        scope = 2;
+        scopeCurator = 2;
+        displayName = Painkillers;
```
Comment (Freddo3000) [inline addons/ace_tweaks/CfgVehicles.hpp:18]: No stringtable?

File: addons/ace_tweaks/ACE_Medical_Treatment_Actions.hpp [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    class Painkillers: Morphine {
+        displayName = "$STR_ARC_Medical_Give_Painkilllers";
+        displayNameProgress = "$STR_ARC_Medical_Giving_Painkilllers";
+        allowSelfTreatment = 1;
+        allowedSelections[] = {"head"};
+        animationMedic = "AinvPknlMstpSnonWnonDnon_medic1";
+        callbackSuccess = "ace_medical_treatment_fnc_medication";
+        category = "medication";
+        condition = "!(_patient getVariable ['ACE_isUnconscious',false])";
+        icon = "\x\arc_misc\addons\ace_tweaks\ui\icons\painkillers.paa";
```
Comment (Freddo3000) [inline addons/ace_tweaks/ACE_Medical_Treatment_Actions.hpp:15]: QPATHTOF

File: addons/ace_tweaks/ACE_Medical_Treatment_Actions.hpp [hunk 10]
```diff
@@ -0,0 +1,21 @@
+
+class ace_medical_treatment_actions {
+
+    // - Syringes -------------------------------------------------------------
+    class Morphine;
+    class Painkillers: Morphine {
+        displayName = "$STR_ARC_Medical_Give_Painkilllers";
```
Comment (Freddo3000) [inline addons/ace_tweaks/ACE_Medical_Treatment_Actions.hpp:7]: CSTRING()

File: addons/ace_tweaks/ACE_Medical_Treatment_Actions.hpp [hunk 11]
```diff
@@ -0,0 +1,21 @@
+
+class ace_medical_treatment_actions {
+
+    // - Syringes -------------------------------------------------------------
+    class Morphine;
+    class Painkillers: Morphine {
```
Comment (Freddo3000) [inline addons/ace_tweaks/ACE_Medical_Treatment_Actions.hpp:6]: GVAR(Painkillers)

File: addons/ace_tweaks/ACE_Medical_Treatment.hpp [hunk 12]
```diff
@@ -0,0 +1,13 @@
+
+class ace_medical_treatment {
+    class Medication {
+        class PainKillers {
```
Comment (Freddo3000) [inline addons/ace_tweaks/ACE_Medical_Treatment.hpp:4]: GVAR(PainKillers)

**Review by Freddo3000 — CHANGES_REQUESTED** [review]
tbh this should probably be a PR to ACE, not here.

**Comment by Drofseh** [comment]
> tbh this should probably be a PR to ACE, not here.

This is actually partially implemented in Ace already, but it was never finished for reasons I don't know.
Will implement suggestions.

File: addons/ace_tweaks/CfgVehicles.hpp (hunk 7 above)
Comment (Drofseh) [reply addons/ace_tweaks/CfgVehicles.hpp:19]: This is a pseudonym I used in a previous unit.

File: addons/ace_tweaks/stringtable.xml (hunk 4 above)
Comment (Drofseh) [reply addons/ace_tweaks/stringtable.xml:4]: What is the right prefix here?
I'm converting from something else to integrate with arc_misc

Comment (Freddo3000) [reply addons/ace_tweaks/stringtable.xml:4]: `STR_arc_misc_ace_tweaks_xyz`, is used with `CSTRING()`, though having a stringtable at all seems a bit overkill for our purposes.

Comment (Freddo3000) [reply addons/ace_tweaks/stringtable.xml:4]: I'd also change project name to `arc_misc` and package name to `ace_tweaks`

File: addons/ace_tweaks/ACE_Medical_Treatment_Actions.hpp (hunk 11 above)
Comment (Drofseh) [reply addons/ace_tweaks/ACE_Medical_Treatment_Actions.hpp:6]: This class is part of the unfinished ace implementation

File: addons/ace_tweaks/ACE_Medical_Treatment.hpp (hunk 12 above)
Comment (Drofseh) [reply addons/ace_tweaks/ACE_Medical_Treatment.hpp:4]: This class is part of the unfinished ace implementation

**Comment by BorderKeeper** [comment]
I think it might be neat for our community to include these unique features that other groups dont have.

**Comment by Freddo3000** [comment]
> I think it might be neat for our community to include these unique features that other groups dont have.

It might be neat, though I don't think it is healthy for the modding community. Think of how many features we'd loose out on if all communities resonated that way?

**Comment by BorderKeeper** [comment]
> > I think it might be neat for our community to include these unique features that other groups dont have.
> 
> It might be neat, though I don't think it is healthy for the modding community. Think of how many features we'd loose out on if all communities resonated that way?

You can see it as us being the testbed needed for the implementation to make it into ACE. We are not usurping features but testing them so it can be obvious that they are a good addition.

**Comment by BorderKeeper** [comment]
This can be PR'd and merged once its suggestion post gets positive reactions.

**Review by BorderKeeper — APPROVED** [review]
Tally from suggestion post on 2nd of December is 9 yes to 1 no. Approved. (https://discord.com/channels/240160552867987475/374416645969543178/916021343567364197)

**Review by WirelessPillow — APPROVED** [review]

**Review by BorderKeeper — APPROVED** [review]
lgtm

**Comment by Freddo3000** [comment]
Decided against implementing on a staff meeting, see https://discord.com/channels/240160552867987475/374416645969543178/916021343567364197

**Pull request closed without merging**